### ISSUE TITLE
windows 11 installation

### ISSUE THREAD

**Issue by CaseyMiller** [brucemiller/LaTeXML#2441]
Installation under windows 11 is failing.  Have tried all suggestions in doc and combed the web.  Tried both the choco and cpan approaches. Have clean install of MikTeX.  Used suggestions in doc for kpsewhich evnironment variables.  cpan build.log file is attached. 

[build.log](https://github.com/user-attachments/files/17935835/build.log)

**Comment by dginev** [comment]
You may be seeing some of the babel-related test regressions on newertexlives, based on the attached log. Those are tracked in issues #2429 and #2434. 

The clean install of MikTeX brings in recent latex toolchains, and latexml is currently experiencing some significant regressions on texlive 2023 and newer, see #2064 as the main tracking issue on that front.

Lastly, latexml does not have an official Windows maintainer, so resolving the OS-specific pieces of this regression (if any) may take a while longer. There is a related Windows regression reported in #2355

**Comment by xworld21** [comment]
@CaseyMiller LaTeXML 0.8.5 is a very old version of LaTeXML. Could you please try installing the latest LaTeXML (0.8.8)?

Unfortunately, as @dginev said, even LaTeXML 0.8.8 can't deal with very recent packages, especially around babel and expl3. If you still get failures, please install using `--notest`.

There may be an issue that needs fixing, though. Please let us know if 0.8.8 is still printing these errors
```
#       got : 'kpsewhich: security risk: running with elevated privileges'
#  expected : 'No obvious problems'
```
These *may* be another instance of #2025, but more likely a simple naive assumption in the test suite. The test should probably only check if the last line is 'No obvious problems' and ignore other messages. I'd need to see the full output of the command being run to confirm.

**Comment by xworld21** [comment]
PS: I just realised that Chocolatey is distributing 0.8.5 as the 'latest' version of LaTeXML. I'll see if I can find the maintainer.

**Comment by CaseyMiller** [comment]
I didn't have any luck installing a new version:

PS C:\Windows\system32> choco install latexml --version 0.8.8
Chocolatey v2.4.0
Installing the following packages:
latexml
By installing, you accept licenses for the packages.
latexml not installed. The package was not found with the source(s) listed.
 Source(s): 'https://community.chocolatey.org/api/v2/'
 NOTE: When you specify explicit sources, it overrides default sources.
If the package version is a prerelease and you didn't specify `--pre`,
 the package may not be found.
Version was specified as '0.8.8'. It is possible that version
 does not exist for 'latexml' at the source specified.
Please see https://docs.chocolatey.org/en-us/troubleshooting for more
 assistance.

Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - latexml - latexml not installed. The package was not found with the source(s) listed.
 Source(s): 'https://community.chocolatey.org/api/v2/'
 NOTE: When you specify explicit sources, it overrides default sources.
If the package version is a prerelease and you didn't specify `--pre`,
 the package may not be found.
Version was specified as '0.8.8'. It is possible that version
 does not exist for 'latexml' at the source specified.
Please see https://docs.chocolatey.org/en-us/troubleshooting for more
 assistance.

**Comment by xworld21** [comment]
> latexml not installed. The package was not found with the source(s) listed.

Chocolatey is stuck at version 0.8.5. The maintainer did not reply, so I guess I'll try a takeover.

You'll have to install LaTeXML by hand. I usually cut it short and tell people to do:
```
cpanm --build-args=CC=c++ --notest --verbose Image::Magick
cpanm --notest --verbose LaTeXML
```
Both ImageMagick and LaTeXML fail tests anyway (IM is out of date on CPAN; LaTeXML does not deal properly with TeX Live >2021).

**Comment by dginev** [comment]
@CaseyMiller if you are interested in a quick resolution, you may be able to install an Ubuntu [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) on your Windows machine and follow the linux instructions.

**Comment by CaseyMiller** [comment]
Great suggestion!

**Comment by brucemiller** [comment]
It appears that the CI tests (on push/merge) running Strawberry Perl are
all failing

**Comment by xworld21** [comment]
@CaseyMiller can you provide logs and error messages and a minimum example of your problem? I run latexml 0.8.8 on Windows 11 every once in a while and it usually works, modulo the MikTeX/TeX Live 2024 incompatibilities. It sounds like we are talking of three different issues:
- Chocolatey is wildly out of date (reported to maintainers)
- tests fail if kpsewhich emits error messages (I'll make a separate issue)
- even when installed, LaTeXML seemingly fails, but we need more details

**Comment by CaseyMiller** [comment]
Here is the build log

cpanm (App::cpanminus) 1.7047 on perl 5.040000 built for MSWin32-x64-multi-thread
Work directory is C:\Users\ctmiller/.cpanm/work/1733921023.59344
You have make C:\Strawberry\c\bin\gmake.exe
You have LWP 6.77
Falling back to Archive::Tar 3.02_001
Searching LaTeXML () on cpanmetadb ...
LaTeXML is up to date. (0.8.8)

The minimal example run with the actual file and output attached
[minimal-example.zip](https://github.com/user-attachments/files/18112958/minimal-example.zip)
:

PS D:\dropbox\research\arc alliance\arc-geophysical-research\jats> latexml --destination=example.html --includestyles article-affiliation-eg.tex
latexml.bat (LaTeXML version 0.8.8) processing article-affiliation-eg.tex
 | Digesting > Processing > Loading > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/textcomp.sty.ltxmlkpsewhich: security risk: running with elevated privileges
kpsewhich: security risk: running with elevated privileges
kpsewhich: security risk: running with elevated privileges
 | Digesting > Processing > Loading > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/keyval.sty.ltxmlkpsewhich: security risk: running with elevated privileges
 | Digesting > Processing > Loading > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/ltxcmds.sty.ltxmlkpsewhich: security risk: running with elevated privileges
 | Digesting > Processing > Loading > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/kvsetkeys.sty.ltxmlkpsewhich: security risk: running with elevated privileges
 | Digesting > Processing > Loading > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/kvdefinekeys.sty.ltxmlkpsewhich: security risk: running with elevated privileges
 | Digesting > Processing > Loading > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/kvoptions.sty.ltxmlkpsewhich: security risk: running with elevated privileges
 / Digesting > Processing > Loading > Loading > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/refcount.sty.ltxmlkpsewhich: security risk: running with elevated privileges
 / Digesting > Processing > Loading > Loading > Loading > Processing > Loading C:/Strawberry/perl/site/lib\LaTeXML/Packkpsewhich: security risk: running with elevated privileges
 / Digesting > Processing > Loading > Loading > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/gettitlestring.sty.kpsewhich: security risk: running with elevated privileges
 -kpsewhich: security risk: running with elevated privilegesocessing definitions C:/Users/ctmiller/AppData/Local/Progray
kpsewhich: security risk: running with elevated privileges
 - Digesting > Processing > Loading > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/bitset.sty.ltxmlkpsewhich: security risk: running with elevated privileges
 - Digesting > Processing > Loading > Loading > Processing > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/intcalkpsewhich: security risk: running with elevated privileges
 - Digesting > Processing > Loading > Loading > Processing > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/bigintkpsewhich: security risk: running with elevated privileges
 \ Digesting > Processing content D:\Dropbox\research\ARC Alliance\ARC-Geophysical-Research\jats\article-affiliation-egkpsewhich: security risk: running with elevated privileges
Error:latex:(kvoptions) Package kvoptions Error: \ProcessLocalKeyvalOptions is intended for packages only         See the kvoptions package documentation for explanation.         I can't help         at doclicense.sty; line 25 col 0
 \ Digesting > Processing > Processing definitions C:/Users/ctmiller/AppData/Local/Programs/MiKTeX/tex/latex/doclicensekpsewhich: security risk: running with elevated privileges
 \ Digesting > Processing > Processing > Processing > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/ifthen.sty.ltkpsewhich: security risk: running with elevated privileges
 \ Digesting > Processing > Processing > Processing definitions C:/Users/ctmiller/AppData/Local/Programs/MiKTeX/tex/latkpsewhich: security risk: running with elevated privileges
 \ Digesting > Processing > Processing definitions C:/Users/ctmiller/AppData/Local/Programs/MiKTeX/tex/latex/doclicensekpsewhich: security risk: running with elevated privileges
 \ Digesting > Processing > Processing > Processing definitions C:/Users/ctmiller/AppData/Local/Programs/MiKTeX/tex/latkpsewhich: security risk: running with elevated privileges
 - Digesting > Processing > Processing definitions C:/Users/ctmiller/AppData/Local/Programs/MiKTeX/tex/latex/doclicensekpsewhich: security risk: running with elevated privileges
Error:undefined:\languagename The token T_CS[\languagename] is not defined. at doclicense.sty; line 278 col 0
 \ Digesting > Processing > Processing definitions C:/Users/ctmiller/AppData/Local/Programs/MiKTeX/tex/latex/doclicensekpsewhich: security risk: running with elevated privileges
kpsewhich: security risk: running with elevated privileges
Warning:latex:(doclicense) Package doclicense Warning: No language definition for \languagename found. Consider adding one and submit it to doclicense. Using English as fallback. at doclicense.sty; line 278 col 0
 \ Digesting > Processing > Processing definitions C:/Users/ctmiller/AppData/Local/Programs/MiKTeX/tex/latex/doclicensekpsewhich: security risk: running with elevated privileges
kpsewhich: security risk: running with elevated privileges
Error:latex:(doclicense) Package doclicense Error: License unknown         See the doclicense package documentation for explanation.         Please check the manual of doclicense at doclicense.sty; line 398 col 15
 \ Digesting > Processing content D:\Dropbox\research\ARC Alliance\ARC-Geophysical-Research\jats\article-affiliation-egkpsewhich: security risk: running with elevated privileges
 \ Digesting > Processing > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/csquotes.sty.ltxmlkpsewhich: security risk: running with elevated privileges
 /kpsewhich: security risk: running with elevated privileges C:/Users/ctmiller/AppData/Local/Programs/MiKTeX/tex/latex/y
 / Digesting > Processing > Loading > Processing > Loading C:/Strawberry/perl/site/lib\LaTeXML/Package/csquotes.def.ltxkpsewhich: security risk: running with elevated privileges
Info:latex:(csquotes)  Package csquotes Info: Trying to load configuration file 'csquotes.cfg'..     at csquotes.sty; line 2484 col 0
 / Digesting > Processing > Loading > Processing definitions C:/Users/ctmiller/AppData/Local/Programs/MiKTeX/tex/latex/kpsewhich: security risk: running with elevated privileges
kpsewhich: security risk: running with elevated privileges
Info:latex:(csquotes)  Package csquotes Info: ... configuration file loaded successfully     at csquotes.sty; line 2487 col 0
 / Digesting > Processing > Loading > Processing definitions C:/Users/ctmiller/AppData/Local/Programs/MiKTeX/tex/latex/kpsewhich: security risk: running with elevated privileges
kpsewhich: security risk: running with elevated privileges
Error:undefined:\doclicenseURL The token T_CS[\doclicenseURL] is not defined. at article-affiliation-eg.tex; line 54 col 0
Conversion complete: 1 warning; 4 errors; 2 undefined macros[\doclicenseURL, \languagename] (See D:/Dropbox/research/ARC Alliance/ARC-Geophysical-Research/jats\article-affiliation-eg.latexml.log) (reqd. 13.59s)

**Comment by xworld21** [comment]
@CaseyMiller it sounds like latexml is working fine. Try the same command with `latexmlc` (reason: latexml generates an xml file that must be converted with latexmlpost; latexmlc does both steps in a single run). If it looks ok, then this issue is just about Chocolatey.

**Comment by CaseyMiller** [comment]
Indeed this is so. The remaining unique issue from this item is that the windows versions are out of date with the installers.

**Comment by xworld21** [comment]
This issue can be closed: the maintainer updated LaTeXML to 0.8.8 on Chocolatey.
